src/routes/news/_news.js

```diff
@@ -0,0 +1,70 @@
+// Dates must be in format YYYY-MM-DD
+// title must be unique within same date
+
+const newsList = [
+    {
+        title: 'ABC',
+        file: '20200207-a.txt',
+        img: 'kolintang.jpg',
+        img_name: 'kolintang',
+        date: '2020-02-07',
+    },
+    {
+        title: 'DEF',
+        file: '20200811.txt',
+        img: 'sasando.jpg',
+        img_name: 'sasando',
+        date: '2020-08-11',
+    },
+    {
+        title: 'GHI',
+        file: '20201113.txt',
+        img: 'angklung.jpg',
+        img_name: 'angklung',
+        date: '2020-11-13',
+    },
+];
+
+const transform = (newsList) => {
+    const monthList = [
+        'January',
+        'February',
+        'March',
+        'April',
+        'May',
+        'June',
+        'July',
+        'August',
+        'September',
+        'October',
+        'November',
+        'December',
+    ];
+    const adaptedNewsList = {};
+
+    for (let news of newsList) {
+        // sanity check if date exists and date follows YYYY-MM-DD format.
+        if (!news.date || !news.date.match(/[0-9]{4}-[0-9]{2}-[0-9]{2}/)) {
+            throw Error('Date format must be in YYYY-MM-DD format');
+        }
+
+        const [year, month, date] = news.date.split('-').map(Number);
+
+        // sanity date checking
+        if (month < 1 || month > 12 || date < 0 || date > 31) {
+            throw Error(`Invalid date ${news.date}`);
+        }
+
+        news.displayDate = `${date} ${monthList[month - 1]} ${year}`;
+        let datesWoSep = news.date.replace(/-/g, '');
```

Comment:
datesWoSep confused me initially, perhaps naming it 'datesWithoutSeparator' will make it much clearer. It's okay to use long name variables.